### PR TITLE
V3 missing client user id filter

### PR DESCRIPTION
**Pull request by Rrathinasabapath**
Please verify and approve this PR for the following change:

- Made changes to Hyperwallet to accommodate the search criterion clientUserId for ListUser method

### CODE REVIEW

File: src/test/resources/integration/listUsers-request.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+curl -X "GET" "https://api.sandbox.hyperwallet.com/rest/v3/users" \
```
Comment (ramahalingam) [inline src/test/resources/integration/listUsers-request.txt:1]: Pls add some query parameters for filtering options here.

File: src/test/java/com/hyperwallet/clientsdk/model/HyperwalletUsersListPaginationOptionsTest.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package com.hyperwallet.clientsdk.model;
+
+import java.util.Date;
```
Comment (ramahalingam) [inline src/test/java/com/hyperwallet/clientsdk/model/HyperwalletUsersListPaginationOptionsTest.java:3]: Pls remove this unused import

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletIT.java [hunk 3]
```diff
@@ -1334,7 +1376,7 @@ public void uploadUserDocuments_invalidDocumentStatus() throws Exception {
 
     //
     // Transfer Refunds
-    //
+    //f
```
Comment (ramahalingam) [inline src/test/java/com/hyperwallet/clientsdk/HyperwalletIT.java:1379]: Typo.

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletIT.java [hunk 4]
```diff
@@ -76,6 +76,48 @@ public void testListUserStatusTransitions() throws Exception {
...
+            mockServer.verify(parseRequest(functionality));
+            throw e;
+        }
+        assertThat(returnValue.getCount(), is(equalTo(2)));
+        HyperwalletUser userOne = returnValue.getData().get(0);
+        assertThat(userOne.getToken(), is(equalTo("usr-00c6cfda-a43d-4b29-a20b-1abc7e800b58")));
+        assertThat(userOne.getCreatedOn(), is(equalTo(dateFormat.parse("2019-10-30T22:15:35 UTC"))));
+        assertThat(userOne.getStatus(), is(equalTo(HyperwalletUser.Status.PRE_ACTIVATED)));
+        assertThat(userOne.getVerificationStatus(), is(equalTo(HyperwalletUser.VerificationStatus.NOT_REQUIRED)));
+    }
```
Comment (ramahalingam) [inline src/test/java/com/hyperwallet/clientsdk/HyperwalletIT.java:119]: Pls add assert statement for the 2nd index as well.

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletUsersListPaginationOptions.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
+package com.hyperwallet.clientsdk.model;
+
+import java.util.Date;
```
Comment (ramahalingam) [inline src/main/java/com/hyperwallet/clientsdk/model/HyperwalletUsersListPaginationOptions.java:3]: Pls remove this unused import

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletIT.java [hunk 6]
```diff
@@ -76,6 +76,48 @@ public void testListUserStatusTransitions() throws Exception {
...
+        initMockServer(functionality);
+
+        HyperwalletList<HyperwalletUser> returnValue;
+        try {
+            returnValue = client.listUsers();
+        } catch (Exception e) {
+            mockServer.verify(parseRequest(functionality));
+            throw e;
+        }
+        assertThat(returnValue.getCount(), is(equalTo(2)));
```
Comment (ramahalingam) [inline src/test/java/com/hyperwallet/clientsdk/HyperwalletIT.java:91]: Pls add assert statement for the 2nd index as well.

**Review by ramahalingam — CHANGES_REQUESTED** [review]
Pls remove all the .DS_store files from this PR.

File: src/test/java/com/hyperwallet/clientsdk/model/HyperwalletUsersListPaginationOptionsTest.java (hunk 2 above)
Comment (Rrathinasabapath) [reply src/test/java/com/hyperwallet/clientsdk/model/HyperwalletUsersListPaginationOptionsTest.java:3]: Done

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletIT.java (hunk 3 above)
Comment (Rrathinasabapath) [reply src/test/java/com/hyperwallet/clientsdk/HyperwalletIT.java:1379]: Done

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletIT.java (hunk 4 above)
Comment (Rrathinasabapath) [reply src/test/java/com/hyperwallet/clientsdk/HyperwalletIT.java:119]: Done

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletUsersListPaginationOptions.java (hunk 5 above)
Comment (Rrathinasabapath) [reply src/main/java/com/hyperwallet/clientsdk/model/HyperwalletUsersListPaginationOptions.java:3]: Done

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletIT.java (hunk 6 above)
Comment (Rrathinasabapath) [reply src/test/java/com/hyperwallet/clientsdk/HyperwalletIT.java:91]: Done

File: src/test/resources/integration/listUsers-request.txt (hunk 1 above)
Comment (Rrathinasabapath) [reply src/test/resources/integration/listUsers-request.txt:1]: As discussed before this couldnt be done as testing dynamically with query parameters is not feasible in the current mock test setup.

**Review by ramahalingam — APPROVED** [review]

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletUsersListPaginationOptions.java [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    public String getProgramToken() {
+        return programToken;
+    }
+
+    public void setProgramToken(String programToken) {
+        this.programToken = programToken;
+    }
+
```
Comment (jbausch-hw) [inline src/main/java/com/hyperwallet/clientsdk/model/HyperwalletUsersListPaginationOptions.java:44]: minor: redundant newLine

**Review by jbausch-hw — APPROVED** [review]

**Review by peter-joseph — APPROVED** [review]

**Review by arao6 — APPROVED** [review]

**Pull request merged**